features/steps/steps.py

```diff
@@ -49,6 +49,22 @@ def fmt(x):
         return v
 
 
+def get_entity_point_components(entity):
+    if entity.is_a('IfcPolyLoop'):
+        points = entity.Polygon
+    elif entity.is_a('IfcPolyline'):
+        points = entity.Points
+    return points
+
+
+def get_sending_system_precision(context):
+    try:
+        geo_representations = context.model.by_type('IFCGEOMETRICREPRESENTATIONCONTEXT')
```

Comment:
There can be multiple rep contexts. I think we should take an effort in connecting the appropriate rep context to the rep item, which is unfortunately not trivial.

```
IfcRepresentation |-> IfcRepresentationContext
                  |-> IfcRepresentationItem (e.g IfcExtrudedAreaSolid) -> ... -> IfcPolyline
```

So you'd need to use ifcopenshell.file.get_inverse() (because there isn't a named inverse from polyline back to where it is referenced) to trace back the representation item chain to the representation and (there could be multiple) and then take the context with the largest precision.

Reply:
Added.

Reply:
I've tested the new solution with a few files, generally works well, but I found some cases, in which an IfcRepresentationContext was not found. An example below:

`#352922=IfcPolyline((#352918,#352920))` -> 
`#352924=IfcGridAxis('11',#352922,.T.) `-> 
`#353156=IfcGrid('1H2s9IBD6pu931S6f7FDpD',#12,'Grid',$,$,#352917,#353153,(#352924,#352942,#352952,#352962,#352972,#353012,#353022,#353032,#353042,#353052,#353062,#353072,#353082,#353092,#353102,#353112,#353122,#353132),(#352982,#352992,#353002,#353142),$)` ->
`#7404=IfcRelContainedInSpatialStructure('04ldtj6cp2dME6CiP80Bzh',#12,$,$,(#7401,#7736`

So starting from `#352922=IfcPolyline((#352918,#352920))` it didn't quite reach an object with Context/Context reference. In the current implementation, a default precision is implemented in that case. Does that make sense or would it be reasonable to perhaps use some other method in addition to get_inverse()?